Next version

hansvancalster opened the pull request:
This PR should address #112 (and some other minor issues)

Steps to do in *protocolsource* repo when this version of protocolhelper is released:

- [x] update protocolhelper in renv.lock
- [x] add ZENODO secret (or ZENODO_SANDBOX in test repo)
- [x] make ZENODO token available in [update_news_zenodo.yml](https://github.com/inbo/protocolsource/blob/main/.github/workflows/update_news_zenodo.yml)
- [x] use `protocolhelper:::update_doi()` in https://github.com/inbo/protocolsource/blob/main/docker/entrypoint_update.sh
- [x] make ZENODO token available in [update_website.yml](https://github.com/inbo/protocolsource/blob/main/.github/workflows/update_website.yml). Nothing needs to be changed in the docker shell because the code to upload to zenodo is integrated in protocolhelper:::render_release() which is already called in that shell.
- [ ] update CONTRIBUTING guide

@ElsLommelen these steps should first be tested in your fork and in that case ZENODO_SANDBOX should be used.

- [ ] discuss with @peterdesmet if an organisation level ZENODO secret can be made for the INBO Zenodo community that can used in these GHA workflows otherwise all protocols will be deposited under my account 
- [ ] decide which way a protocol should be submitted to a community. Currently, the code uses zenodo$submitRecordToCommunities(record, communities), but see https://cran.r-project.org/web/packages/zen4R/vignettes/zen4R.html#ZenodoDeposit-communities for an alternative

ElsLommelen commented:
> @ElsLommelen these steps should first be tested in your fork and in that case ZENODO_SANDBOX should be used.

I'm not sure if this makes sense. as I remember there was a problem with secrets in forks: a PAT cannot be used or at least reacts differently in GHA in a fork than in the main repo.  That's why we tested everything in a separate repo [testprotocol](https://github.com/ElsLommelen/testprotocol) when testing our GHA.

In this repo testprotocol I tried updating protocolhelper in renv using [this tutorial](https://tutorials.inbo.be/tutorials/r_renv_update/) (as a package developer I never need/use it), but despite I installed the new branch using `remotes::install_github("inbo/protocolhelper@next-version")`, the last commit of the main branch is added to `renv.lock`.  I pushed this trial to [this branch](https://github.com/ElsLommelen/testprotocol/tree/test-next-version), feel free to go on with it.

hansvancalster commented:
> > @ElsLommelen these steps should first be tested in your fork and in that case ZENODO_SANDBOX should be used.
> 
> I'm not sure if this makes sense. as I remember there was a problem with secrets in forks: a PAT cannot be used or at least reacts differently in GHA in a fork than in the main repo. That's why we tested everything in a separate repo [testprotocol](https://github.com/ElsLommelen/testprotocol) when testing our GHA.

Yes indeed, that was my intention but I was not clear.

> 
> In this repo testprotocol I tried updating protocolhelper in renv using [this tutorial](https://tutorials.inbo.be/tutorials/r_renv_update/) (as a package developer I never need/use it), but despite I installed the new branch using `remotes::install_github("inbo/protocolhelper@next-version")`, the last commit of the main branch is added to `renv.lock`. I pushed this trial to [this branch](https://github.com/ElsLommelen/testprotocol/tree/test-next-version), feel free to go on with it.

I think I have pushed commits to your branch that address this. You should have the next-version of protocolhelper if you pull these commits and execute `renv::restore()`.

ElsLommelen commented:
For the Zenodo secret: is there a ZENODO_SANDBOX available already, or should I add this first (and how)?

ElsLommelen commented:
And some more questions to be sure I do it correctly at the first try:
- should the Zenodo token be made available for the step 'update protocol', or elsewhere as well?
- where exactly should `protocolhelper::update_doi()` be added in the workflow?

hansvancalster commented:
> For the Zenodo secret: is there a ZENODO_SANDBOX available already, or should I add this first (and how)?

Can you give me admin rights to testprotocol. I will add it.

hansvancalster commented:
> And some more questions to be sure I do it correctly at the first try:
> 
> * should the Zenodo token be made available for the step 'update protocol', or elsewhere as well?
> * where exactly should `protocolhelper::update_doi()` be added in the workflow?

- [ ] make ZENODO_SANDBOX token available in [update_news_zenodo.yml](https://github.com/ElsLommelen/testprotocol/blob/main/.github/workflows/update_news_zenodo.yml).  This file is currently missing in the testprotocol repo. Copy it from here: https://github.com/inbo/protocolsource/blob/main/.github/workflows/update_news_zenodo.yml
- [ ] use `protocolhelper:::update_doi()` in https://github.com/ElsLommelen/testprotocol/blob/main/docker/entrypoint_update.sh. Insert it on line 23. 
- [ ] make ZENODO_SANDBOX token available in [update_website.yml](https://github.com/ElsLommelen/testprotocol/blob/main/.github/workflows/update-website.yml).

ElsLommelen commented:
> > For the Zenodo secret: is there a ZENODO_SANDBOX available already, or should I add this first (and how)?
> 
> Can you give me admin rights to testprotocol. I will add it.

I don't think it is possible to add roles or rights to persons in personal repositories.  I added you as a collaborator in the past, but unlike INBO repositories, it is not possible to choose a role here.

ElsLommelen commented:
> [ ] make ZENODO_SANDBOX token available in [update_news_zenodo.yml](https://github.com/ElsLommelen/testprotocol/blob/main/.github/workflows/update_news_zenodo.yml). This file is currently missing in the testprotocol repo. Copy it from here: https://github.com/inbo/protocolsource/blob/main/.github/workflows/update_news_zenodo.yml

Yes, I know, I already copied this file to my repo (locally), but as far as I can see, the token is not yet added to this file.  And that's why I asked where to add the token in the file.  As most steps don't need it, I suppose it is on the step 'update protocol', but also there the secret is not used (even not after adding the suggestion of the next step).

ElsLommelen commented:
> I think I had something like above in mind. This should make the token available in the entrypoint_update.sh where it will be used by protocolhelper:::update_doi(). Not sure it will work. That is why we are testing it.
> 
> EDIT: see https://github.com/inbo/protocolhelper/blob/d26ef289ce574d9601b26111feff097b1e270447/R/update_doi.R#L19C1-L27C2

I suppose I should add `update_doi(token = secrets.ZENODO_SANDBOX)` to pass the secret?  The keyring-stuff might work locally, but I don't see how it could possibly get a secret in the docker container that runs in the GHA (where there is no connection to your local computer).

hansvancalster commented:
> > I think I had something like above in mind. This should make the token available in the entrypoint_update.sh where it will be used by protocolhelper:::update_doi(). Not sure it will work. That is why we are testing it.
> > EDIT: see https://github.com/inbo/protocolhelper/blob/d26ef289ce574d9601b26111feff097b1e270447/R/update_doi.R#L19C1-L27C2
> 
> I suppose I should add `update_doi(token = secrets.ZENODO_SANDBOX)` to pass the secret? The keyring-stuff might work locally, but I don't see how it could possibly get a secret in the docker container that runs in the GHA (where there is no connection to your local computer).

No it should still work on GHA. See https://keyring.r-lib.org/index.html#github

ElsLommelen commented:
Also, when I add this function, what do you expect as a result?  Don't forget this is run in a docker container, so we need to make information available somehow before this container is closed and removed again.  For a container in a GHA I see 2 possibilities for this:  either print output that will be visible in the logs (similar to the sessioninfo that is printed), or save a file and commit it in the repo (similar to the .zenodo.json or the NEWS.md).

hansvancalster commented:
> Also, when I add this function, what do you expect as a result? Don't forget this is run in a docker container, so we need to make information available somehow before this container is closed and removed again. For a container in a GHA I see 2 possibilities for this: either print output that will be visible in the logs (similar to the sessioninfo that is printed), or save a file and commit it in the repo (similar to the .zenodo.json or the NEWS.md).

Yes indeed, it is similar. The function update_doi() will add or update the doi field in the YAML section of index.Rmd. So the result should be a changed index.Rmd that is committed and pushed to the PR branch in a similar way as this is done for .zenodo.json or NEWS.md. So something like:

```
echo '\nUpdating doi in index.Rmd...\n'
Rscript --no-save --no-restore -e 'protocolhelper:::update_doi(protocol_code = "'$GITHUB_HEAD_REF'", sandbox = TRUE)'
git add index.Rmd
git commit --message="update doi in index.Rmd"

```

Needs to be inserted at line 23.

The arguments `protocol_code` and `path` are taken from  `update_news_release()` function, so they should be correct in the above code. Also `sandbox = TRUE` is the default, but I have added this explicitly so we do not forget to change this later.

ElsLommelen commented:
Done in [PR 86 in testprotocol](https://github.com/ElsLommelen/testprotocol/pull/86).

ElsLommelen reviewed (commented):
I had a quick look at the most recent changes and added some minor suggestions.  To be sure everything is covered, you might want to have a look at issue #112 and tick everything that has been solved?
Finally I also noticed some changed that we made specifically for testing reasons, so it seems a good idea to change these already with the final version in mind, even though it won't work yet.  At least if you will update protocolsource in a short time.

hansvancalster commented on data-raw/code_for_internal_data.R:
```diff
@@ -2,54 +2,65 @@
 
 # code to prepare reserved_codes internal dataset goes here
 library(googlesheets4)
+library(janitor)
```
It's not needed. I just run that file locally to update internal package data.

hansvancalster commented on NEWS.md:
```diff
@@ -1,3 +1,16 @@
+
+## Bug fixes
+
+* Authors were not rendered correctly in PDF version.
```
Will change this

hansvancalster commented on .github/workflows/check_on_branch.yml:
```diff
@@ -14,11 +14,12 @@ jobs:
     env:
       CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      ZENODO_SANDBOX: ${{ secrets.ZENODO_SANDBOX }}
```
No, this should remain sandbox. It is only for needed here for unit testing of protocolhelper functions.

hansvancalster commented on .github/workflows/check_on_different_r_os.yml:
```diff
@@ -30,6 +30,8 @@ jobs:
       _R_CHECK_SYSTEM_CLOCK_: false
       RSPM: ${{ matrix.config.rspm }}
       GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      ZENODO_SANDBOX: ${{ secrets.ZENODO_SANDBOX }}
```
Idem. Only for unit testing, so should stay sandbox.

hansvancalster commented on .github/workflows/check_on_main.yml:
```diff
@@ -15,5 +15,6 @@ jobs:
     env:
       CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      ZENODO_SANDBOX: ${{ secrets.ZENODO_SANDBOX }}
```
Idem. Should stay sandbox.

hansvancalster commented on .github/workflows/check_on_branch.yml:
```diff
@@ -14,11 +14,12 @@ jobs:
           pandoc-version: '3.1.11'
 
-      - uses: inbo/actions/check_pkg@main
+      - uses: inbo/actions/check_pkg@devel
```
I will change this back to main. Can you start a PR to update the protocols actions in inbo/actions repo (referring to ElsLommelen/actions to see what needs to be changed)?

ElsLommelen reviewed (changes requested):
Ik ga het goedkeuren, maar terwijl ik door de checks scrolde, vielen me wel enkele dings op waardoor ik me afvraag of checklist wel werkt als verwacht:

- lintr geeft een error omwille van het ontbrekende package janitor, maar dit wordt blijkbaar genegeerd 
![image](https://github.com/user-attachments/assets/f23e5e6e-b95f-48b7-9baf-073e178e928d)

- En bij de code coverage staat onderaan dit (ik weet niet wat het betekent, mogelijk een tijdelijke uitval van de website op het moment dat je runde?):  
![image](https://github.com/user-attachments/assets/c6a0840b-d1f7-4852-9d79-ee426104826c)

ElsLommelen reviewed (approved):
Sorry, blijkbaar onbedoeld op een verkeerd knopje geduwd bij een poging om de 'submit review' knop te bereiken (die net onder m'n scherm zat)
